examples/validation/_01_validation.py

```diff
@@ -0,0 +1,126 @@
+
+In this example, we will learn how to use the :func:`~tpcp.validate.validate` function implemented in
+tcpc.
+For this, we will reuse the pipeline and data from the example on :ref:` gridsearch <grid_search>`.
+If you want to have more information on how the dataset and pipeline is built, head over to this example.
+Here we will just copy the code over.
+"""
+# %%
+# Dataset
+from pathlib import Path
+
+import numpy as np
+
+from examples.datasets.datasets_final_ecg import ECGExampleData
+
+try:
+    HERE = Path(__file__).parent
+except NameError:
+    HERE = Path(".").resolve()
+data_path = HERE.parent.parent / "example_data/ecg_mit_bih_arrhythmia/data"
+example_data = ECGExampleData(data_path)
+
+# %%
+# Pipeline
+import pandas as pd
+
+from examples.algorithms.algorithms_qrs_detection_final import QRSDetector
+from tpcp import Parameter, Pipeline, cf
+
+
+class MyPipeline(Pipeline[ECGExampleData]):
+    algorithm: Parameter[QRSDetector]
+
+    r_peak_positions_: pd.Series
+
+    def __init__(self, algorithm: QRSDetector = cf(QRSDetector())):
+        self.algorithm = algorithm
+
+    def run(self, datapoint: ECGExampleData):
+        # Note: We need to clone the algorithm instance, to make sure we don't leak any data between runs.
+        algo = self.algorithm.clone()
+        algo.detect(datapoint.data["ecg"], datapoint.sampling_rate_hz)
+
+        self.r_peak_positions_ = algo.r_peak_positions_
+        return self
+
+
+# %%
+# The Scorer
+# ----------
+# The scorer is identical to the scoring function used in the other examples.
+# The F1-score is still the most important parameter for our comparison.
+from examples.algorithms.algorithms_qrs_detection_final import match_events_with_reference, precision_recall_f1_score
+
+
+def score(pipeline: MyPipeline, datapoint: ECGExampleData):
+    # We use the `safe_run` wrapper instead of just run. This is always a good idea.
+    pipeline = pipeline.safe_run(datapoint)
+    tolerance_s = 0.02  # We just use 20 ms for this example
+    matches = match_events_with_reference(
+        pipeline.r_peak_positions_.to_numpy(),
+        datapoint.r_peak_positions_.to_numpy(),
+        tolerance=tolerance_s * datapoint.sampling_rate_hz,
+    )
+    precision, recall, f1_score = precision_recall_f1_score(matches)
+    return {"precision": precision, "recall": recall, "f1_score": f1_score}
+
+
+# %%
+# Validation
+# ----------------
+# Now we have all the pieces for the final validation.
+# First we need to create instances of our data and pipeline.
+# Finally, we can call `tpcp.validate.validate`.
+from tpcp.validate import validate
+
+pipe = MyPipeline()
+
+results = validate(pipe, example_data, scoring=score)
+result_df = pd.DataFrame(results)
+result_df
+
+
+# %%
+# Understanding the Results
+# -------------------------
+# The cross validation provides a lot of outputs.
+# To simplify things a little, we will split the output into three parts:
+#
+# The main output are the dataset performance values.
```

Comment:
Maybe specify that by default these are the "means" over all datapoints.
If you need other aggreagtions -> see the custom scorer example